step-automation-packages/step-automation-packages-controller/src/main/java/step/automation/packages/AutomationPackageExecutionPlugin.java

```diff
@@ -0,0 +1,36 @@
+package step.automation.packages;
+
+import step.core.execution.ExecutionContext;
+import step.core.plugins.IgnoreDuringAutoDiscovery;
+import step.core.plugins.Plugin;
+import step.engine.plugins.AbstractExecutionEnginePlugin;
+
+@IgnoreDuringAutoDiscovery
+@Plugin
+public class AutomationPackageExecutionPlugin extends AbstractExecutionEnginePlugin {
+
+    private final static String AUTOMATION_PACKAGE_LOCK = "$automation.package.lock";
+    private final AutomationPackageLocks automationPackageLocks;
+
+    public AutomationPackageExecutionPlugin(AutomationPackageLocks automationPackageLocks) {
+        this.automationPackageLocks = automationPackageLocks;
+    }
+
+    @Override
+    public void executionStart(ExecutionContext context) {
+        super.executionStart(context);
+        String automationPackageId = (String) context.getPlan().getCustomField(AutomationPackageEntity.AUTOMATION_PACKAGE_ID);
+        if (automationPackageId != null) {
+            automationPackageLocks.readLock(automationPackageId);
```

Comment:
- Would add a comment to explain the goal of thie plugin or at least of this line. 
- Do we want to wait indefinitely? not sure if should have a timeout...

Reply:
Added comments and timeout to acquire the read lock (default 2 minutes).